gcloud-java-storage/src/test/java/com/google/gcloud/storage/StorageImplTest.java

```diff
@@ -0,0 +1,830 @@
+    EasyMock.expect(storageRpcMock.list(BUCKET_LIST_OPTIONS)).andReturn(result);
+    EasyMock.replay(storageRpcMock);
+    ListResult<BucketInfo> listResult = storage.list(BUCKET_LIST_MAX_RESULT, BUCKET_LIST_PREFIX);
+    assertEquals(cursor, listResult.nextPageCursor());
+    assertArrayEquals(bucketList.toArray(), Iterables.toArray(listResult, BucketInfo.class));
+  }
+
+  @Test
+  public void testListBlobs() {
+    EasyMock.expect(optionsMock.storageRpc()).andReturn(storageRpcMock).times(2);
+    EasyMock.expect(optionsMock.retryParams()).andReturn(RetryParams.noRetries());
+    EasyMock.replay(optionsMock);
+    storage = StorageFactory.instance().get(optionsMock);
+    String cursor = "cursor";
+    ImmutableList<BlobInfo> blobList = ImmutableList.of(BLOB_INFO1, BLOB_INFO2);
+    Tuple<String, Iterable<com.google.api.services.storage.model.StorageObject>> result =
+        Tuple.of(cursor, Iterables.transform(blobList,
+            new Function<BlobInfo, com.google.api.services.storage.model.StorageObject>() {
+              @Override
+              public StorageObject apply(BlobInfo f) {
+                return f.toPb();
+              }
+            }));
+    EasyMock.expect(storageRpcMock.list(BUCKET_NAME1, EMPTY_RPC_OPTIONS)).andReturn(result);
+    EasyMock.replay(storageRpcMock);
+    ListResult<BlobInfo> listResult = storage.list(BUCKET_NAME1);
+    assertEquals(cursor, listResult.nextPageCursor());
+    assertArrayEquals(blobList.toArray(), Iterables.toArray(listResult, BlobInfo.class));
+  }
+
+  @Test
+  public void testListBlobsEmpty() {
+    EasyMock.expect(optionsMock.storageRpc()).andReturn(storageRpcMock).times(2);
+    EasyMock.expect(optionsMock.retryParams()).andReturn(RetryParams.noRetries());
+    EasyMock.replay(optionsMock);
+    storage = StorageFactory.instance().get(optionsMock);
+    EasyMock.expect(storageRpcMock.list(BUCKET_NAME1, EMPTY_RPC_OPTIONS))
+        .andReturn(
+            Tuple.<String, Iterable<com.google.api.services.storage.model.StorageObject>>of(null,
+                null));
+    EasyMock.replay(storageRpcMock);
+    ListResult<BlobInfo> listResult = storage.list(BUCKET_NAME1);
+    assertEquals(null, listResult.nextPageCursor());
+    assertArrayEquals(ImmutableList.of().toArray(), Iterables.toArray(listResult, BlobInfo.class));
+  }
+
+  @Test
+  public void testListBlobsWithOptions() {
+    EasyMock.expect(optionsMock.storageRpc()).andReturn(storageRpcMock).times(2);
+    EasyMock.expect(optionsMock.retryParams()).andReturn(RetryParams.noRetries());
+    EasyMock.replay(optionsMock);
+    storage = StorageFactory.instance().get(optionsMock);
+    String cursor = "cursor";
+    ImmutableList<BlobInfo> blobList = ImmutableList.of(BLOB_INFO1, BLOB_INFO2);
+    Tuple<String, Iterable<com.google.api.services.storage.model.StorageObject>> result =
+        Tuple.of(cursor, Iterables.transform(blobList,
+            new Function<BlobInfo, com.google.api.services.storage.model.StorageObject>() {
```

Comment:
Here and in other places. Any reason not to use the package scope functions defined in [BlobInfo](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobInfo.java#L54).